website/content/docs/k8s/observability/metrics.mdx

```diff
@@ -0,0 +1,115 @@
+```yaml
+connectInject:
+  metrics:
+    defaultMergedMetricsPort: 20100
+```
+
+The endpoint to scrape service metrics from can be configured only on a per-Pod basis via the Pod annotations `consul.hashicorp.com/service-metrics-port` and `consul.hashicorp.com/service-metrics-path`. If these are not configured, the service metrics port will default to the port used to register the service with Consul (`consul.hashicorp.com/connect-service-port`), which in turn defaults to the first port on the first container of the Pod. The service metrics path will default to `/metrics`.
+
+## Consul Agent Metrics
+
+Metrics from the Consul server and client Pods can be scraped via Prometheus by setting the field `global.metrics.enableAgentMetrics` to `true`. Additionally, one can configure the metrics retention time on the agents by configuring
+the field `global.metrics.agentMetricsRetentionTime` which expects a duration and defaults to `"1m"`. This value must be greater than `"0m"` for the Consul servers and clients to emit metrics at all. As the Prometheus deployment currently does not support scraping TLS endpoints, agent metrics are currently *unsupported* when TLS is enabled.
+
+```yaml
+global:
+  metrics:
+    enabled: true
+    enableAgentMetrics: true
+    agentMetricsRetentionTime: "1m"
+```
+
+## Gateway Metrics
+
+Metrics from the Consul gateways, namely the Ingress Gateway, Terminating Gateway and the Mesh Gateway can be scraped via Prometheus by setting the field `global.metrics.enableGatewayMetrics`. The gateways emit standard Envoy proxy metrics. To ensure that the metrics are not exposed to the public internet (as Mesh and Ingress gateways can have public IPs), their metrics endpoints are exposed on the Pod IP of the respective gateway instance, rather than on all interfaces on `0.0.0.0`.
+
+```yaml
+global:
+  metrics:
+    enabled: true
+    enableGatewayMetrics: true
+```
+
+## Metrics in the UI Topology Visualization
+
+Consul’s built in UI has a topology visualization for services part of the Consul-Connect service mesh. The topology visualization has the ability to fetch basic metrics from a metrics provider for each service and display those metrics as part of the [topology visualization](/docs/connect/observability/ui-visualization).
+
+The diagram below illustrates how the UI displays service metrics for a sample application:
+
+[![UI Topology View](/img/ui-topology.png)](/img/ui-topology.png)
+
+This could previously be configured via `server.extraConfig` but can now be configured under `ui.metrics`. This will enable the Consul UI to query the provider specified by `ui.metrics.provider` at the URL of the Prometheus server `ui.metrics.baseURL` to display
+service metrics. When `global.metrics.enabled` and `ui.enabled` are both set to true, the UI will display Prometheus metrics.
+
+```yaml
```

Comment:
Should this snippet show a case when you want to override some of the values or set `ui.metrics.enabled` to false? It might be a bit confusing to readers because they might expect the yaml snippet to show how to enable this feature, but in this case, there's nothing that you would need to set in your helm config to enable it.

Reply:
I reworded so it would show how to enable this feature. Thanks for catching this!